pkg/queue/instance.go

```diff
@@ -116,13 +131,17 @@ func (q *queueImpl) processNextItem() bool {
 	}
 
 	// Run the task.
-	if err := task(); err != nil {
+	if err := (*task)(); err != nil {
 		delay := q.delay
 		log.Infof("Work item handle failed (%v), retry after delay %v", err, delay)
 		time.AfterFunc(delay, func() {
-			q.Push(task)
+			q.Push(*task)
 		})
 	}
+	if q.metrics != nil {
+		q.metrics.done(task)
```

Comment:
we probably should do this even in event of an error

Reply:
Sure, I can do that.

Reply:
Sorry, if I am missing something, but if there is an error it will be recorded in the same flow as there is no early return in case of error and `ime.AfterFunc` is non-blocking.